native/src/main/java/io/ballerina/stdlib/grpc/Message.java

```diff
@@ -756,13 +761,26 @@ private String filterDescAnnotation(BMap<BString, Object> annotations) {
     }
 
     private boolean isDescriptorAnnotationAvailable(RecordType recordType) {
-        return Arrays.stream(recordType.getAnnotations().getKeys()).anyMatch(
-                s -> isValidProtoAnnotation(s.getValue()));
+        return Arrays.stream(recordType.getAnnotations().getKeys()).anyMatch(s -> isValidProtoAnnotation(s.getValue()));
     }
 
     private boolean isValidProtoAnnotation(String annotationName) {
-        return annotationName.contains("ballerina") && annotationName.contains("protobuf") &&
-                annotationName.contains("Descriptor");
+        return annotationName.contains(ORG_NAME) && annotationName.contains("protobuf") &&
+                annotationName.contains(ANN_DESCRIPTOR);
+    }
+
+    private Descriptors.Descriptor getMessageDescriptorFromMessageRegistry(String messageName) {
+        Map<String, Descriptors.Descriptor> messageDescriptorMap = MessageRegistry.getInstance()
+                .getMessageDescriptorMap();
+        if (messageDescriptorMap.containsKey(messageName)) {
+            return messageDescriptorMap.get(messageName);
+        }
+        for (String key : messageDescriptorMap.keySet().toArray(new String[]{})) { // for messages with a package name
+            if (key.endsWith("." + messageName)) {
```

Comment:
this has some limitations where there are two messages with the same name and different packages. but this is not applied to the old implementations.

Reply:
Update in https://github.com/ballerina-platform/module-ballerina-grpc/pull/961/commits/8028f5629a466af37f4feb0668ae1998f4d324e5